deploy/helm/pulumi-operator/templates/role.yaml

```diff
@@ -0,0 +1,90 @@
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Any way to sync these templates with the configs in deploy/crds/yaml? Should we just make this the default way to specify the roles etc.? I just worry about drift over time since these now need to be specified in two places. Perhaps helm is the right way to specify it and allow users to run `helm template` if someone needs to get at the raw templates?